### PR TITLE
Reduce Logging Level

### PR DESCRIPTION
**Pull request by Maithem**
## Overview
Reduce StreamAddressSpace trim log from warn to trace.

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

**Comment by zhangn49** [comment]
Happened to see this log is flooding in a setup. Why it gets called so many times?

**Comment by Maithem** [comment]
> Happened to see this log is flooding in a setup. Why it gets called so many times?
> 
> > 2021-02-25T20:17:14.724Z  WARN netty-0 StreamAddressSpace - trim: attempting to trim non-valid address -1
> > 2021-02-25T20:17:14.776Z  WARN netty-0 StreamAddressSpace - trim: attempting to trim non-valid address -1
> > 2021-02-25T20:17:14.827Z  WARN netty-0 StreamAddressSpace - trim: attempting to trim non-valid address -1
> > 2021-02-25T20:17:14.878Z  WARN netty-0 StreamAddressSpace - trim: attempting to trim non-valid address -1
> > 2021-02-25T20:17:14.930Z  WARN netty-0 StreamAddressSpace - trim: attempting to trim non-valid address -1
> > 2021-02-25T20:17:14.981Z  WARN netty-0 StreamAddressSpace - trim: attempting to trim non-valid address -1
> > 2021-02-25T20:17:15.032Z  WARN netty-0 StreamAddressSpace - trim: attempting to trim non-valid address -1
> > 2021-02-25T20:17:15.084Z  WARN netty-0 StreamAddressSpace - trim: attempting to trim non-valid address -1

While I was fixing bitset related bug I added the log line here : 
https://github.com/CorfuDB/CorfuDB/pull/2866/files#diff-2316e4c0466f66804dc3be7894fc4a9263114393d28bafc7be324c59fb71c3f7R165

File: runtime/src/main/java/org/corfudb/runtime/view/stream/StreamAddressSpace.java [hunk 1]
```diff
@@ -162,7 +162,7 @@ private void removeAddresses(List<Long> addresses) {
     public void trim(long trimMark) {
         if (!Address.isAddress(trimMark)) {
             // If not valid address return and do not attempt to trim.
-            log.warn("trim: attempting to trim non-valid address {}", trimMark);
+            log.trace("trim: attempting to trim non-valid address {}", trimMark);
```
Comment (zhangn49) [inline runtime/src/main/java/org/corfudb/runtime/view/stream/StreamAddressSpace.java:165]: should be good to wrapped with if log.isTraceEnabled?

**Review by zhangn49 — APPROVED** [review]

**Review by annym — APPROVED** [review]

**Review by apavan — APPROVED** [review]

File: runtime/src/main/java/org/corfudb/runtime/view/stream/StreamAddressSpace.java (hunk 1 above)
Comment (Maithem) [reply runtime/src/main/java/org/corfudb/runtime/view/stream/StreamAddressSpace.java]: Done.

**Review by pankti-m — APPROVED** [review]

**Comment by apavan** [comment]
@Maithem Injection framework would be waste of resource and time. Please help force merge this change after integration tests are completed.

File: runtime/src/main/java/org/corfudb/runtime/view/stream/StreamAddressSpace.java (hunk 1 above)
Comment (xnull) [reply runtime/src/main/java/org/corfudb/runtime/view/stream/StreamAddressSpace.java]: Actually "parametrized messages" were added to slf4j not to pollute code with endless if(log.isTraceEnabled)
Please stop using if conditions for logs http://www.slf4j.org/faq.html#logging_performance

**Pull request merged**